mithril-client-wasm/src/client_wasm.rs

```diff
@@ -88,15 +91,64 @@ impl MithrilClient {
                 .map_err(|err| format!("Failed to parse options: {err:?}"))
                 .unwrap()
         };
-        let unstable = client_options.unstable;
-        let client = ClientBuilder::aggregator(aggregator_endpoint, genesis_verification_key)
-            .add_feedback_receiver(feedback_receiver)
-            .with_options(client_options)
+        let mut client_builder =
+            ClientBuilder::aggregator(aggregator_endpoint, genesis_verification_key)
+                .add_feedback_receiver(feedback_receiver)
+                .with_options(client_options.clone());
+
+        let certificate_verifier_cache = if client_options.unstable
+            && client_options.enable_certificate_chain_verification_cache
+        {
+            let cache = Self::build_certifier_cache(
+                aggregator_endpoint,
+                TimeDelta::seconds(
+                    client_options.certificate_chain_verification_cache_duration_in_seconds as i64,
+                ),
+            );
+            if let Some(cache) = &cache {
```

Comment:
Isn't possible to make this treatment on the result variable `certificate_verifier_cache` ? 
We can then bring closer `client_builer` declaration and this treatment.

Reply:
We could yes !
One thing that I wonder is: should the `with_certificate_verifier_cache` of the builder be able to take an option ? In a way we could think of this as a possibility to disable the cache at the client api level and this would hugely simplify the construction of the builder here (since it would avoid to have to store the builder in a variable).